CMakeLists.txt

```diff
@@ -63,31 +63,48 @@ endif ()
 # GoogleTest                                                                                      #
 # =============================================================================================== #
 
-# Download and unpack googletest
+# Search for GoogleTest, and fallback to downloading it if not found
 if (ZYCORE_BUILD_TESTS)
-    if (NOT DEFINED ZYCORE_DOWNLOADED_GTEST)
-        configure_file("CMakeLists.txt.in" "${CMAKE_BINARY_DIR}/gtest/download/CMakeLists.txt")
-        execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
-            RESULT_VARIABLE result
-            WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/gtest/download")
-        if (result)
-            message(FATAL_ERROR "CMake step for googletest failed: ${result}")
-        endif()
-        execute_process(COMMAND ${CMAKE_COMMAND} --build .
-            RESULT_VARIABLE result
-            WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/gtest/download")
-        if (result)
-            message(FATAL_ERROR "Build step for googletest failed: ${result}")
-        endif()
-
-        set(ZYCORE_DOWNLOADED_GTEST TRUE CACHE BOOL "")
-        mark_as_advanced(ZYCORE_DOWNLOADED_GTEST)
-    endif ()
+    find_package(GTest QUIET)
+    if (GTest_FOUND)
+        # CMake 3.20 and upstream GTestConfig.cmake
+        if (TARGET GTest::gtest)
+            add_library(gtest ALIAS GTest::gtest)
+        # CMake 3.5 and newer
+        elseif (TARGET GTest::GTest)
+            add_library(gtest ALIAS GTest::GTest)
+        # Older CMake versions
+        else ()
+            add_library(gtest INTERFACE)
+            find_package(Threads REQUIRED)
+            target_include_directories(gtest INTERFACE "${GTEST_INCLUDE_DIRS}")
+            target_link_libraries(gtest INTERFACE "${GTEST_LIBRARIES}" Threads::Threads)
+        endif ()
```

Comment:
Like with the other comment, feel free to just raise the required CMake version. The oldest non-EOL Ubuntu version (18.04) has CMake 3.10.2, oldest non-EOL Debian (buster) has 3.13.4. Since these two distros are typically the slowest to upgrade, it's pretty safe to assume that all other important ones have newer versions already. I'm not particularly concerned with the commercial extended support releases.